mlflow/pytorch.py

```diff
@@ -0,0 +1,144 @@
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+def _load_model(path, **kwargs):
+    mlflow_model_path = os.path.join(path, "MLmodel")
+    if not os.path.exists(mlflow_model_path):
+        raise RuntimeError("MLmodel is not found at '{}'".format(path))
+
+    mlflow_model = Model.load(mlflow_model_path)
+
+    if FLAVOR_NAME not in mlflow_model.flavors:
+        raise ValueError("Could not find flavor '{}' amongst available flavors {}, "
+                         "unable to load stored model"
+                         .format(FLAVOR_NAME, list(mlflow_model.flavors.keys())))
+
+    # This maybe replaced by a warning and then try/except torch.load
+    flavor = mlflow_model.flavors[FLAVOR_NAME]
+    if torch.__version__ != flavor["pytorch_version"]:
+        raise ValueError("Unfortunately stored model version '{}' does not match "
+                         "installed PyTorch version '{}'"
+                         .format(flavor["pytorch_version"], torch.__version__))
+
+    path = os.path.abspath(path)
+    path = os.path.join(path, mlflow_model.flavors[FLAVOR_NAME]['model_data'])
+    return torch.load(path, **kwargs)
+
+
+def load_model(path, run_id=None, **kwargs):
+    """
+    Load a PyTorch model from a local file (if run_id is None) or a run.
+    :param path: Local filesystem path or Run-relative artifact path to the model saved 
+                 by `mlflow.pytorch.log_model`.
+    :param run_id: Run ID. If provided it is combined with path to identify the model.
+    :param kwargs: kwargs to pass to `torch.load` method
+    """
+    if run_id is not None:
+        path = mlflow.tracking._get_model_log_dir(model_name=path, run_id=run_id)
+
+    return _load_model(path, **kwargs)
+
+
+def load_pyfunc(path, **kwargs):
+    """
+    Load the model as PyFunc. The loaded PyFunc exposes a `predict(pd.DataFrame) -> pd.DataFrame`
+    method that, given an input DataFrame of n rows and k float-valued columns, feeds a
+    corresponding (n x k) torch.FloatTensor as input to the PyTorch model. ``predict`` returns
+    the model's predictions (output tensor) in a single-column DataFrame.
+
+    :param path: Local filesystem path to the model saved by `mlflow.pytorch.log_model`.
+    :param kwargs: kwargs to pass to `torch.load` method
+    """
+    return _PyTorchWrapper(_load_model(os.path.dirname(path), **kwargs))
+
+
+class _PyTorchWrapper(object):
+    """
+    Wrapper class that creates a predict function such that
+    predict(data: pd.DataFrame) -> model's output as pd.DataFrame (pandas DataFrame)
+    """
+    def __init__(self, pytorch_model):
+        self.pytorch_model = pytorch_model
+
+    def predict(self, data, device='cpu'):
+        if not isinstance(data, pd.DataFrame):
+            raise TypeError("Input data should be pandas.DataFrame")
+        self.pytorch_model.eval()
+        with torch.no_grad():
+            input_tensor = torch.from_numpy(data.values.astype(np.float32)).to(device)
```

Comment:
I think I have a small correction to add here. We can have an incoherence between `device` and `**kwargs` of `load_pyfunc(path, **kwargs)`. Namely if model is loaded on GPU and than user sets device as CPU and vice-versa.